src/_edge-functions/referer.js

```diff
@@ -0,0 +1,42 @@
+import { HTMLRewriter } from "https://ghuc.cc/worker-tools/html-rewriter/index.ts";
+
+const messages = {
+  "localhost:8888": `✨ Happy developing ✨`,
+  "netlify.com": `🎉 You found my site via netlify.com! <a href="/blog/add-personalization-to-static-html-with-edge-functions-no-browser-javascript/">Learn how to show this banner using Netlify Edge Functions</a>`,
+  "reddit.com": "👀 Hello there, Redditor! I see you. Please be nice.",
+};
+
+export default async (request, context) => {
+  // get HTTP referer header
+  const referer = request.headers.get("referer");
+
+  // get the next HTTP response in the chain
+  const response = await context.next();
+
+  // if no referer, continue
+  if (referer === null) {
+    return response;
+  }
+
+  // get keys from refererMessages
+  const refererKeys = Array.from(Object.keys(messages));
+
+  // look through keys, and check to see if the incoming referrer contains one of the values
+  const findRefererKey = refererKeys.filter((ref) => referer.includes(ref)).pop();
+
+  // if we don't have a referer match, return the response
+  if (findRefererKey === undefined) {
+    return response;
+  }
+
+  // if we do have a referer match, rewrite the element
+  // in the DOM with a friendly message
```

Comment:
Tiny thing... we're transforming the HTML which is being sent in the response, rather than updating the DOM which I associate with the browser API which interprets and models the HTML.

I only mention this due to the desire to be really clear that this transformation is happening before the browser ever gets wind of any HTML. 

Perhaps a tweak such as:

```suggestion
  // in the response HTML with a friendly message
```